OpenStack-Rabbit-Consumer/usr/local/bin/message_consumer.py

```diff
@@ -101,22 +185,38 @@ def consume(message):
             vm_name = message.get("payload").get("display_name")
             username = message.get("_context_user_name")
             metadata = message.get("payload").get("metadata")
+            machinename = message.get("payload").get("metadata").get("AQ_MACHINENAME")
 
-            logger.info("Project Name: %s (%s)" % (project_name, project_id))
-            logger.info("VM Name: %s (%s) " % (vm_name, vm_id))
-            logger.info("Username: " + username)
-            logger.info("Hostnames: %s" % metadata.get('HOSTNAMES'))
+            logger.info("Project Name: %s (%s)", project_name, project_id)
+            logger.info("VM Name: %s (%s) ", vm_name, vm_id)
+            logger.info("Username: %s", username)
+            logger.info("Hostnames: %s", metadata.get('HOSTNAMES'))
+            for host in metadata.get("HOSTNAMES").split(","):
+                try:
+                    aq_api.delete_host(host)
+                except Exception as e:
+                    logger.error("Failed to delete host: %s", e)
+                    openstack_api.update_metadata(project_id,
+                        vm_id, {"AQ_STATUS" : "FAILED"})
+                    raise Exception("Failed to delete host")
+
+            try:
+                aq_api.delete_machine(machinename)
+            except Exception as e:
+                raise Exception("Failed to delete machine")
 
             try:
                 for host in metadata.get('HOSTNAMES').split(','):
-                    aq_api.vm_delete(host)
+                    aq_api.reset_env(host,machinename)
             except Exception as e:
-                logger.error("Failed to reset Aquilon configuration: " + str(e))
-                openstack_api.update_metadata(project_id, vm_id, {"AQ_STATUS" : "FAILED"})
+                logger.error("Failed to reset Aquilon configuration: %s", e)
+                openstack_api.update_metadata(project_id,
+                    vm_id, {"AQ_STATUS" : "FAILED"})
                 raise Exception("Failed to reset Aquilon configuration")
 
             logger.info("Successfully reset Aquilon configuration")
-            logger.info("=== Finished Aquilon deletion hook for VM " + vm_name + " ===")
+            logger.info("=== Finished Aquilon deletion hook for VM %s ===",
+                vm_name)
 
 
 def on_message(channel, method, header, raw_body):
```

Comment:
`header` doesn't appear to be used, if you need to have an argument to implement a callback, you can use `_` as the argument name, but better if you can remove it.